### ISSUE TITLE
[New Rule] match with same arms

### ISSUE THREAD

**Issue by sbrugman** [MartinThoma/flake8-simplify#194]
## Explanation

Implement the checking for branches with identical arm bodies for `match`
Same benefits as https://github.com/MartinThoma/flake8-simplify/issues/10 

## Example

```python
# Bad
match hello:
   case "world":
        print("world")
    case "foobar":
       print("world")

# Good

match hello:
   case "world" | "foobar":
       print("world")
```
